x-pack/plugins/security_solution/server/endpoint/lib/artifacts/lists.ts

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { createHash } from 'crypto';
+import { validate } from '../../../../common/validate';
+
+import {
+  Entry,
+  EntryNested,
+  EntryMatch,
+  EntryMatchAny,
+} from '../../../../../lists/common/schemas/types/entries';
+import { FoundExceptionListItemSchema } from '../../../../../lists/common/schemas/response/found_exception_list_item_schema';
+import { ExceptionListClient } from '../../../../../lists/server';
+import {
+  InternalArtifactSchema,
+  TranslatedEntry,
+  TranslatedEntryMatch,
+  TranslatedEntryMatchAny,
+  TranslatedEntryNested,
+  WrappedTranslatedExceptionList,
+  wrappedExceptionList,
+} from '../../schemas';
+import { ArtifactConstants } from './common';
+
+export async function buildArtifact(
+  exceptions: WrappedTranslatedExceptionList,
+  os: string,
+  schemaVersion: string
+): Promise<InternalArtifactSchema> {
+  const exceptionsBuffer = Buffer.from(JSON.stringify(exceptions));
+  const sha256 = createHash('sha256').update(exceptionsBuffer.toString()).digest('hex');
+
+  return {
+    identifier: `${ArtifactConstants.GLOBAL_ALLOWLIST_NAME}-${os}-${schemaVersion}`,
+    sha256,
+    encoding: 'application/json',
+    created: Date.now(),
+    body: exceptionsBuffer.toString('base64'),
+    size: exceptionsBuffer.byteLength,
+  };
+}
+
+export async function getFullEndpointExceptionList(
+  eClient: ExceptionListClient,
+  os: string,
+  schemaVersion: string
+): Promise<WrappedTranslatedExceptionList> {
+  const exceptions: WrappedTranslatedExceptionList = { exceptions_list: [] };
+  let numResponses = 0;
+  let page = 1;
+
+  do {
+    const response = await eClient.findExceptionListItem({
+      listId: 'endpoint_list',
```

Comment:
We should move this out into a common place because we'll need to use it in the UI too, I think